### DISCUSSION TITLE
New Plugin available - Key Replacement plugin.

### DISCUSSION THREAD

**Discussion by bjazmoore** [GetPublii/Publii#1721] [Useful Resources]
**What is it?**

This is a plugin that allows you to create key/replacement text pairs in the plugin interface (adding as many as you would like).  During render when the key is located it is replaced with the replacement text.

**Why**

Well - I have wanted it for my own uses, I just have not gotten around to learning about plugins until now.  I saw someone else struggling with this idea also, so I guess the timing is right.

**Where?**

Get it from my repository where there is also some basic usage instructions as well: 

[**github.com/bjazmoore/Key-Replacement**](https://github.com/bjazmoore/Key-Replacement)

Thanks

**Comment by candidexmedia** [comment]
Fantastic, thank you!

**Comment by dziudek** [comment]
Great! For simple replacements sounds like good solution :) For more advanced usage - with regexps, replacements limited to specific parts of website, replacements outside the post content but also in site structure it is worth to consider use: https://marketplace.getpublii.com/plugins/smart-replacer/

**Reply by bjazmoore** [reply]
> Great! For simple replacements sounds like good solution :) For more advanced usage - with regexps, replacements limited to specific parts of website, replacements outside the post content but also in site structure it is worth to consider use: https://marketplace.getpublii.com/plugins/smart-replacer/

Absolutely.  This is not meant to replace the Smart Replacer which does significantly more especially supporting regex replacements.  Thanks

**Reply by dma2386** [reply]
BUT... The Smart Replacer Plugin documentation states... "The Smart Replacer Plugin is a powerful tool for automating text and regular expression replacements throughout your Publii website." - This implies that it works in exactly the same way as MS Word Find and replace: specify the text to replace, then specify the text that is to replace that text. - But it doesn't work that way. It replaces the text in preview mode,"it applies transformations to the generated output" but not the actual text in the post - which is what's required. It isn't a  Publii 'Global Find and Replace' plugin, which it seems to be described as, and which would be very very useful.

**Comment by Vinc26** [comment]
Great, thanks!!!
